cmd/maya-apiserver/spc-watcher/handler.go

```diff
@@ -70,7 +70,13 @@ func (c *Controller) spcEventHandler(operation string, spcGot *apis.StoragePoolC
 		// CreateStoragePool function will create the storage pool
 		// It is a create event so resync should be false and pendingPoolcount is passed 0
 		// pendingPoolcount is not used when resync is false.
-		err := CreateStoragePool(spcGot, false, 0)
+		var newSpcLease Leases
+		newSpcLease = &spcLease{spcGot, SpcLeaseKey, c.clientset}
+		_, err := newSpcLease.GetLease()
+		if err != nil {
+			return addEvent, fmt.Errorf("could not acquire lease on spc object:%v", err)
+		}
```

Comment:
@sonasingh46 , can you once check the function processNextWorkItem() from runner.go and its return type. I'm trying to figure out where the error is finally logged and it seems it has to be logged inside processNextWorkItem().

Reply:
sure

Reply:
```
// Run the syncHandler, passing it the namespace/name string of the
		//SPC resource to be synced.
		if err := c.syncHandler(q.Key, q.Operation, q.Object); err != nil {
			return fmt.Errorf("Error syncing '%s': %s", q.Key, err.Error())
		}
		glog.V(1).Infof("Successfully synced '%s'", q.Key)
		return nil
	}(obj)

	if err != nil {
		runtime.HandleError(err)
		return true
	}
```
In above snippet , error is not logged, it is handled runtime in if block.
Logging will cause a flood. But if you say, i can put a level log there inside the if block.
@avishnu

Reply:
fixed.... by logging the error